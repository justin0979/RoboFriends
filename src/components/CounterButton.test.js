import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

it('should render CounterButton component', () => {
  const mockColor = 'red';
  expect(shallow(<CounterButton color={mockColor} />).debug()).toMatchSnapshot();
});

it('correctly increments the counter', () => {
  const mockColor = 'red';

  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('#counter').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
  expect(wrapper.props()).toHaveProperty('color', 'red');
});
