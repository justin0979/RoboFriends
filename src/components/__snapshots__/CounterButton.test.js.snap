// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CounterButton component 1`] = `
"<button id=\\"counter\\" color=\\"red\\" onClick={[Function]}>
  Count: 
  0
</button>"
`;
